mono/metadata/icall.c

```diff
@@ -3185,9 +3190,10 @@ ves_icall_InternalExecute (MonoReflectionMethod *method, MonoObject *this_arg, M
 				MonoClassField* field = mono_class_get_field_from_name (k, str);
 				if (field) {
 					MonoClass *field_klass =  mono_class_from_mono_type (field->type);
-					if (field_klass->valuetype)
-						result = mono_value_box (domain, field_klass, (char *)this_arg + field->offset);
-					else 
+					if (field_klass->valuetype) {
+						result = mono_value_box_checked (domain, field_klass, (char *)this_arg + field->offset, &error);
+						mono_error_set_pending_exception (&error);
```

Comment:
Took me a while to figure out wth it's doing and it's only been a week: I'll add a comment that it's falling through to get to the `g_free`.